# HG changeset patch
# User Sean Farley <sean@mcs.anl.gov>
# Date 1322301160 21600
#      Sat Nov 26 03:52:40 2011 -0600
# Node ID 08d452d6944e316a09839f4777fec96c6456b671
# Parent  b8b0cc4abb9bed53f909e0ac38b29d9f76875236
make: add rule for shared library

This works by putting libmumps_common objects into the other libs.

diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -60,20 +60,25 @@ libseqneeded:
 
 # Build the libpord.a library and copy it into $(topdir)/lib
 $(libdir)/libpord$(PLAT)$(LIBEXT):
 	if [ "$(LPORDDIR)" != "" ] ; then \
 	  cd $(LPORDDIR); \
-	  $(MAKE) CC="$(CC)" CFLAGS="$(OPTC)" AR="$(AR)" RANLIB="$(RANLIB)" OUTC=$(OUTC) LIBEXT=$(LIBEXT); \
+	  $(MAKE) CC="$(CC)" CFLAGS="$(OPTC)" AR="$(AR)" RANLIB="$(RANLIB)" PLAT=$(PLAT) OUTC=$(OUTC) LIBEXT=$(LIBEXT) SHAREDLIBEXT=$(SHAREDLIBEXT) SHAREDFLAGS="$(SHAREDFLAGS)" PREFIX=$(PREFIX); \
 	fi;
 	if [ "$(LPORDDIR)" != "" ] ; then \
-	  cp $(LPORDDIR)/libpord$(LIBEXT) $@; \
+		if [ -s "$(LPORDDIR)/libpord$(PLAT)$(LIBEXT)" ] ; then \
+			cp $(LPORDDIR)/libpord$(PLAT)$(LIBEXT) $(libdir)/libpord$(PLAT)$(LIBEXT); \
+	  fi; \
+		if [ -s "$(LPORDDIR)/libpord$(PLAT)$(SHAREDLIBEXT)" ] ; then \
+	  	cp $(LPORDDIR)/libpord$(PLAT)$(SHAREDLIBEXT) $(libdir)/libpord$(PLAT)$(SHAREDLIBEXT); \
+	  fi; \
 	fi;
 
 clean:
 	(cd src; $(MAKE) clean)
 	(cd examples; $(MAKE) clean)
-	(cd $(libdir); $(RM) *$(PLAT)$(LIBEXT))
+	(cd $(libdir); $(RM) *$(PLAT)$(LIBEXT) *$(PLAT)$(SHAREDLIBEXT))
 	(cd libseq; $(MAKE) clean)
 	if [ $(LPORDDIR) != "" ] ; then \
 	  cd $(LPORDDIR); $(MAKE) realclean; \
         fi;
 
diff --git a/PORD/lib/Makefile b/PORD/lib/Makefile
--- a/PORD/lib/Makefile
+++ b/PORD/lib/Makefile
@@ -19,14 +19,15 @@ OBJS = graph.o gbipart.o gbisect.o ddcre
 # original SPACE package.
 
 .c.o:
 	$(CC) $(COPTIONS) -c $*.c $(OUTC)$*.o
 
-libpord$(LIBEXT):$(OBJS)
+libpord$(PLAT)$(LIBEXT):$(OBJS)
 	$(AR)$@ $(OBJS)
 	$(RANLIB) $@
+	$(CC) $(OBJS) $(CFLAGS) $(SHAREDFLAGS) -install_name $(PREFIX)/lib/libpord$(PLAT)$(SHAREDLIBEXT) -o libpord$(PLAT)$(SHAREDLIBEXT)
 
 clean:
 	rm -f *.o
 
 realclean:
-	rm -f *.o libpord.a
+	rm -f *.o libpord$(LIBEXT) libpord$(SHAREDLIBEXT)
diff --git a/examples/Makefile b/examples/Makefile
--- a/examples/Makefile
+++ b/examples/Makefile
@@ -17,32 +17,32 @@ c:	csimpletest
 z:	zsimpletest
 
 
 include $(topdir)/Makefile.inc
 
-LIBMUMPS_COMMON = $(libdir)/libmumps_common$(PLAT)$(LIBEXT)
+LIBMUMPS_COMMON = $(libdir)/libmumps_common$(PLAT)$(SHAREDLIBEXT)
 
 
-LIBSMUMPS = $(libdir)/libsmumps$(PLAT)$(LIBEXT) $(LIBMUMPS_COMMON)
+LIBSMUMPS = $(libdir)/libsmumps$(PLAT)$(SHAREDLIBEXT) $(LIBMUMPS_COMMON)
 
 ssimpletest:  $(LIBSMUMPS)  $$@.o
 	$(FL) -o $@ $(OPTL) ssimpletest.o  $(LIBSMUMPS) $(LORDERINGS) $(LIBS) $(LIBBLAS) $(LIBOTHERS)
 
 
-LIBDMUMPS = $(libdir)/libdmumps$(PLAT)$(LIBEXT) $(LIBMUMPS_COMMON)
+LIBDMUMPS = $(libdir)/libdmumps$(PLAT)$(SHAREDLIBEXT) $(LIBMUMPS_COMMON)
 
 dsimpletest: $(LIBDMUMPS)  $$@.o 
 	$(FL) -o $@ $(OPTL) dsimpletest.o  $(LIBDMUMPS) $(LORDERINGS) $(LIBS) $(LIBBLAS) $(LIBOTHERS)
 
 
-LIBCMUMPS = $(libdir)/libcmumps$(PLAT)$(LIBEXT) $(LIBMUMPS_COMMON)
+LIBCMUMPS = $(libdir)/libcmumps$(PLAT)$(SHAREDLIBEXT) $(LIBMUMPS_COMMON)
 
 csimpletest: $(LIBCMUMPS)  $$@.o
 	$(FL) -o $@ $(OPTL) csimpletest.o  $(LIBCMUMPS) $(LORDERINGS) $(LIBS) $(LIBBLAS) $(LIBOTHERS)
 
 
-LIBZMUMPS = $(libdir)/libzmumps$(PLAT)$(LIBEXT) $(LIBMUMPS_COMMON)
+LIBZMUMPS = $(libdir)/libzmumps$(PLAT)$(SHAREDLIBEXT) $(LIBMUMPS_COMMON)
 
 zsimpletest: $(LIBZMUMPS)  $$@.o
 	$(FL) -o $@ $(OPTL) zsimpletest.o  $(LIBZMUMPS) $(LORDERINGS) $(LIBS) $(LIBBLAS) $(LIBOTHERS)
 
 
@@ -55,23 +55,23 @@ c_example:	$(LIBDMUMPS) $$@.o
 	$(FC) $(OPTF) $(INCS) -I. -I$(topdir)/include -c $*.F $(OUTF)$*.o
 .c.o:
 	$(CC) $(OPTC) $(INCS) -I. -I$(topdir)/include -c $*.c $(OUTC)$*.o
 
 
-$(libdir)/libsmumps$(PLAT)$(LIBEXT):
+$(libdir)/libsmumps$(PLAT)$(SHAREDLIBEXT):
 	@echo 'Error: you should build the library' $@ 'first'
 	exit -1
 
-$(libdir)/libdmumps$(PLAT)$(LIBEXT):
+$(libdir)/libdmumps$(PLAT)$(SHAREDLIBEXT):
 	@echo 'Error: you should build the library' $@ 'first'
 	exit -1
 
-$(libdir)/libcmumps$(PLAT)$(LIBEXT):
+$(libdir)/libcmumps$(PLAT)$(SHAREDLIBEXT):
 	@echo 'Error: you should build the library' $@ 'first'
 	exit -1
 
-$(libdir)/libzmumps$(PLAT)$(LIBEXT):
+$(libdir)/libzmumps$(PLAT)$(SHAREDLIBEXT):
 	@echo 'Error: you should build the library' $@ 'first'
 	exit -1
 
 $(LIBMUMPS_COMMON):
 	@echo 'Error: you should build the library' $@ 'first'
diff --git a/src/Makefile b/src/Makefile
--- a/src/Makefile
+++ b/src/Makefile
@@ -53,14 +53,16 @@ OBJS = 	$(ARITH)mumps_part1.o\
 
 
 $(libdir)/libmumps_common$(PLAT)$(LIBEXT):	$(OBJS_COMMON)
 	$(AR)$@ $?
 	$(RANLIB) $@
+	$(FC) $? $(OPTF) $(SHAREDFLAGS) $(LORDERINGS) $(SCALAP) $(libdir)/libpord$(SHAREDLIBEXT) -install_name $(PREFIX)/lib/libmumps_common$(PLAT)$(SHAREDLIBEXT) -o $(libdir)/libmumps_common$(PLAT)$(SHAREDLIBEXT)
 
 $(libdir)/lib$(ARITH)mumps$(PLAT)$(LIBEXT):    $(OBJS)
 	$(AR)$@ $?
 	$(RANLIB) $@
+	$(FC) $? $(OPTF) $(SHAREDFLAGS) $(LORDERINGS) $(SCALAP) $(libdir)/libpord$(SHAREDLIBEXT) $(OBJS_COMMON) -install_name $(PREFIX)/lib/lib$(ARITH)mumps$(PLAT)$(SHAREDLIBEXT) -o $(libdir)/lib$(ARITH)mumps$(PLAT)$(SHAREDLIBEXT)
 
 $(ARITH)mumps_load.o:		$(ARITH)mumps_comm_buffer.o \
 				$(ARITH)mumps_struc_def.o
 
 $(ARITH)mumps_ooc.o: 		$(ARITH)mumps_struc_def.o \
